doc/setup_instructions.md

```diff
@@ -84,12 +84,23 @@ desktop development featureset, that should have everything needed for
 C# development. Please be patient with this as this is going to take
 tens of minutes to install and take a lot of disk space.
 
-It might also be possible to make things work with Visual studio code
-as well, but there are no instructions for that. Note: that Godot
-requirements say that visual studio build tools are enough but they
-may not include nuget, which is needed, so if you go that route you
-may need to manually install nuget.
+It is also be possible to make things work with visual studio code
+as well, but the setup process is more complicated and more likely to not work. Note: that Godot requirements say that visual studio build tools are enough but they may not include nuget, which is needed, so if you go that route you may need to manually install nuget.
 
+### Visual Studio Code
+To setup visual studio code to work with Godot, you'll need to have visual studio code, where you can install it here: https://code.visualstudio.com/
+
+Note: Setting up with visual studio code is possible in Linux, but is not recommended, use MonoDevelop instead.
+
+Next, install build tools for visual studio here: https://visualstudio.microsoft.com/downloads/?q=build+tools For the installation process, make sure MSBuild tools is listed under Installation details, it should be there by default.
+
+Go to https://dotnet.microsoft.com/download Under the .NET Core section, click on _Download .NET Core SDK_. Run the installer. Go back to the main download page and find _All .NET Framework Downloads_ under the .NET framework section. Install 4.7 and select the Developer Pack.
+
+On visual studio code, go to the Extensions tab. Get the extensions _C#_, _Mono Debug_, and _godot-tools_.
+
+Open up a new Godot Project. On the top, go to Editor -> Editor Settings. Scroll to the bottom, where you'll find Mono. Click on Editor and set External Editor to visual studio code. Click on Builds and set Build Tool to MSBuild (VS Build Tools).
+
+If you want to setup live debugging with Godot, on the top, go to Project -> Project Settings. Near the bottom, you will find Mono. Click on Debugger Agent. When you want to use the Debugger, turn the Wait for Debugger on. Set the Wait Timeout to how many milliseconds you want Godot to wait for visual studio code debugging, recommended to be set around 15000. Copy the port number and go to visual studio code. Make sure visual studio code is open to your Godot folder. Go to the debug tab and click on _create a launch.json file_. Select C# Mono. When the _launch.json_ file is opened, change the port number to the one you copied previously. Save the file and you can close it. On the Debug tab, switch Launch to Attach. Now, whenever you want to debug, make sure Wait for Debugger is on in Godot, run the project, and press the green arrow in the debugger in visual studio code.
```

Comment:
Could you add new lines to the text so that it is only around 80 columns wide

Reply:
Ok, will change that